### ISSUE TITLE
Old-style context definition should only be interpreted as read context

### ISSUE THREAD

**Issue by bongoman** [GothenburgBitFactory/taskwarrior#2620]
#### To report a bug...

Defining a context using "or" operators in 2.6.0 will then write the "or" values into tasks added with that context set.

https://www.reddit.com/r/taskwarrior/comments/q2vb0f/comment/hfs9d9f/?utm_source=share&utm_medium=web2x&context=3

* What command(s) did you run?

Let's say I define a "mine" context:

`task context define mine "+personal or +home or +work"`

and then switch to that context:

`task context mine`

and then add a task:

`task add this is my task`

When I then run `task`, I see the newly added task has prepended multiple instances of  'or' from the context definition to the description, one for each tag added to the task:

`or or this is my task`

**Comment by tom-doerr** [comment]
This is what I get when I run `task add test` in context `h`:
```
Name          Value                                              
ID            1182                                               
Description   ( ( -admin -sm -lrz -hobby -ing -uni ) or or ) test
Status        Pending                                            
Entered       2021-10-15 02:00:04 (1min)
Last modified 2021-10-15 02:00:04 (1min)                         
Tags          next scheduled_today_custom
Virtual tags  LATEST ORPHAN PENDING READY TAGGED UDA UNBLOCKED   
UUID          a247ba77-8a28-4a5a-874d-bf7465a59ffa
Urgency       17.7                                               
Ti            -

    tags             0.9 *    1 =    0.9
    UDA priority.      1 *  1.8 =    1.8
    TAG next           1 *   15 =     15
                                  ------
                                    17.7

```
`.taskrc`:
```
# Theme
include ~/.task/dark-green-256.theme
color.tagged=
color.tag.next_local=rgb055
color.tag.clarify=rgb515


# Change priority values 
uda.priority.values = H,M,,L
urgency.uda.priority.L.coefficient=-8
urgency.uda.priority..coefficient=1.8
urgency.active.coefficient=50
urgency.user.tag.next_local.coefficient=10

# Contexts
context.ingroups=+ingroups or +ing or pro:ing. or pro:ingroups.
context.hobby=+hobby or +_hobby or pro:hobby.
context.arzt=+arzt or pro:arzt.
context.bu=+bu or +_bu
context.shallow=+shallow or +sh or pro:shallow.
context.ba=+b or +_b or +ba or +_ba or pro:ba.
context.task=+task or pro:task.
context.wg=+wg or pro:wg.
context.admin=+admin or pro:admin.
context.inf=+inf -prof  or pro:inf.
context.leute=+leute
context.mering=+mering  or pro:mering.
context.lrz=+lrz or pro:lrz.
context.sm=+sm or pro:sm.
context.stat=+stat or pro:stat.
context.theo=+theo or pro:theo.
context.master=+master or pro:master.
context.ai=+ai
context.mt=+mt
context.idl=+idl
context.gmm=+gmm
context.vwl=+vwl
context.nlp=+nlp
context.ml=+ml
context.gki=+gki
context.padm=+padm
context.rbad=+rbad
context.dlcg=+dlcg
context.ad=+ad
context.mlr=+mlr
context.adlr=+adlr
context.qc=+qc
context.warp=+warp
context.adlc=+adlc
context.ea=+ea
context.uni=+uni
context.prof=+prof
context.fp=+fp
context.flh=+flh
context.ma=+ma
context.dmkd=+dmkd
context.mvf=+mvf
context.review=-WAITING -sm -bu
context.cs=+cs
context.agt=+agt
context.rladv=+rladv
context.ta=+ta

## Home
context.h=((-admin -sm -lrz -hobby -ing -uni) or +next or +scheduled_today_custom)

## Default not home
context.nh=((-admin -sm -lrz -h -home -hobby -ing -uni) or ((+next or +scheduled_today_custom) and -h and -home))




# UDAs
uda.estimate.type=duration
uda.estimate.label=Est

uda.time_of_day.type=string
uda.time_of_day.label=Ti
uda.time_of_day.default=-

uda.soft_deadline.type=date
uda.soft_deadline.label=SD


report.timeboxing.description=Report that shows tasks scheduled for today with the time estimate
report.timeboxing.columns=id,description,project,due.remaining,estimate,time_of_day
report.timeboxing.filter=((+scheduled_today_custom or -CHILD) and due.before:now+24h) and status:Pending
# report.timeboxing.sort=due+\/,urgency-
report.timeboxing.sort=time_of_day+

report.report1.description=My standard report for viewing tasks
report.report1.columns=id,description,project,until,estimate
report.report1.filter=status:Pending -BLOCKED and (-ing or +self) and -bu -sm
report.report1.sort=urgency-


report.review_bucket_items.description=Report for reviewing bucket items
report.review_bucket_items.columns=id,start.age,tags,description
report.review_bucket_items.filter=+bu +PENDING
report.review_bucket_items.sort=id+


report.list.columns=ID,Active,Age,D,P,Project,Tags,R,Sch,Due,Until,Description,Urg,Est,Scheduled
_forcecolor=on

search.case.sensitive=no

uda.reviewed.type=date
uda.reviewed.label=Reviewed
report._reviewed.description=Tasksh review report.  Adjust the filter to your needs.
report._reviewed.columns=uuid
report._reviewed.sort=reviewed+,modified+
#report._reviewed.filter=( reviewed.none: or reviewed.before:now-6days ) and ( +PENDING or +WAITING ) and (-bu or +sm) and -PARENT
report._reviewed.filter=( reviewed.none: or reviewed.before:now-6days ) and +PENDING and -WAITING and (-bu or +sm) and -PARENT

bulk=3
confirmation=no
nag=
recurrence.confirmation=no
context.ai.vim_codex= +ai +vim_codex
context.ai.vim_codex.subproject1= +ai +vim_codex +subproject1
context.ai.vim_codex.paper= +ai +vim_codex +paper
context.ai.vim_codex.readme= +ai +vim_codex +readme
context.ai.vim_codex.temperature= +ai +vim_codex +temperature
context.dating= +dating
context.uni.lernplan_skript=lege skript +uni +lernplan_skript
context.ssh_verbindungsgeschwindigkeit= +ssh_verbindungsgeschwindigkeit
context.leute.veranstaltungen_frauen=+leute +leute +veranstaltungen_frauen
context.hif= +hif
context.hif.creatine= +hif +creatine
context.task.beeminder_zeit_bis_5= +task +beeminder_zeit_bis_5
context.buch= +buch
context.buch.mk= +buch +mk
context.grosse_kaffee_maschine_saeubern= +grosse_kaffee_maschine_saeubern
context.ai.dl_input= +ai +dl_input
context.ai.zsh_codex= +ai +zsh_codex
context.fuer_veranstaltungen_registrieren= +fuer_veranstaltungen_registrieren
context=h

# Disable context set message.
message.context.set=

```

**Comment by tom-doerr** [comment]
It works for me to just overwrite the context using `task add rc.context=none `... .

**Comment by tbabej** [comment]
@tom-doerr That's disabling the context altogether, however.

Your situation is that you're using old-style context definitions, please run `task news` to see how the configuration needs to change, it's all explained there :slightly_smiling_face: (TL;DR you need to define `context.<name>.read` and `context.<name>.write` and remove the old-context definition)

**Comment by tbabej** [comment]
This got me thinking, however, that the migration path right now is not according to the principle of least surprise - what would be more in line with user expectations is the old-style context definition would be only interpreted as a read context :thinking: I'll spin this up into another issue.
